File: ios/Konnecta/Utils/ShareUtils.swift
import SwiftUI
import UIKit

struct ShareSheet: UIViewControllerRepresentable {
    let activityItems: [Any]
    
    func makeUIViewController(context: Context) -> UIActivityViewController {
        UIActivityViewController(activityItems: activityItems, applicationActivities: nil)
    }
    
    func updateUIViewController(_ uiViewController: UIActivityViewController, context: Context) {}
}

extension View {
    func shareActivity(activity: ActivityWithParticipants) {
        let timeText = activity.start_time != nil ? " a les \(activity.start_time!)" : ""
        let text = """
            Ei! Estem organitzant això per KONNECTA:
            
            *\(activity.title.uppercased())*
            📅 \(activity.day_of_week.uppercased()), \(activity.weekend_date)\(timeText)
            
            Anima't i apunta't a l'app!
        """
        
        let av = UIActivityViewController(activityItems: [text], applicationActivities: nil)
        
        if let scene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
           let rootVC = scene.windows.first?.rootViewController {
            rootVC.present(av, animated: true)
        }
    }
}

File: android/app/src/main/kotlin/com/konnecta/app/utils/ShareUtils.kt
package com.konnecta.app.utils

import android.content.Context
import android.content.Intent
import com.konnecta.app.data.model.*

object ShareUtils {
    fun shareActivity(context: Context, activity: ActivityWithParticipants) {
        val timeText = if (activity.start_time != null) " a les ${activity.start_time}" else ""
        val text = """
            Ei! Estem organitzant això per KONNECTA:
            
            *${activity.title.uppercase()}*
            📅 ${activity.day_of_week.uppercase()}, ${activity.weekend_date}$timeText
            
            Anima't i apunta't a l'app!
        """.trimIndent()

        val sendIntent: Intent = Intent().apply {
            action = Intent.ACTION_SEND
            putExtra(Intent.EXTRA_TEXT, text)
            type = "text/plain"
        }

        val shareIntent = Intent.createChooser(sendIntent, null)
        context.startActivity(shareIntent)
    }
}
